deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -56,7 +56,10 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         results = []
         classes = dataset.classes
         label = cast(pd.Series, dataset.label_col)
-        n_samples = label.groupby(label).count()
+        if context.with_display:
+            n_samples = label.groupby(label).count()
+        else:
+            n_samples = dict(zip(classes, [None] * len(classes)))
```

Comment:
weird that n_samples is a dict, consider changing the name

Reply:
It's a series in the with_display==True case, but a dict has the same api